[Client Feature] Forge System (Fiendish and Influenced Creatures from 12.80)

murilo09 opened the pull request:
# Description

## Implementation of the Forge
### New features:
- Added talkactions for use in forge system on [talkactions/god/forge_functions.lua](https://github.com/opentibiabr/canary/pull/543/files#diff-ed69e6d53c82c3c15ee506b7523448d8ff020477fe935cb774051e49ebd0fe0a)
Talkactions "/adddusts, /removedusts, /getdusts, /setdusts"
Talkaction to teleport to fiendish creature "/fiendish" and for influenced "/influenced"
Talkaction to set monster fiendish "/setfiendish", need look to monster direction and use talkaction
Added in "/m" talkaction the param for create monster fiendish/influenced, example:
"/m rat, 1" create an influenced Rat with level 1 (allowed is 1-5)
"/m rat, fiendish" create a Rat with fiendish status
Talkaction to open forge "/openforge"

- Added spell to find fiendish "exiva moe res"

NOTE: The system is, as far as possible, the same as the global one, there is still a lot of information about the system that nobody knows.

New database table for forge system:
```sql
CREATE TABLE `forge_history` (
    `id` int NOT NULL AUTO_INCREMENT,
    `player_id` int NOT NULL,
    `action_type` int NOT NULL DEFAULT '0',
    `description` text NOT NULL,
    `is_success` tinyint NOT NULL DEFAULT '0',
    `bonus` tinyint NOT NULL DEFAULT '0',
    `done_at` bigint NOT NULL,
    `done_at_date` datetime DEFAULT NOW(),
    `cost` bigint UNSIGNED NOT NULL DEFAULT '0',
    `gained` bigint UNSIGNED NOT NULL DEFAULT '0',
    CONSTRAINT `forge_history_pk` PRIMARY KEY (`id`),
    FOREIGN KEY (`player_id`) REFERENCES `players` (`id`) ON DELETE CASCADE
) ENGINE='InnoDB' DEFAULT CHARSET=utf8;
```
## Missing features

  - [x] Fusion
  - [x] Transfer
  - [x] Conversion
  - [x] Forge History

## Type of change

  - [X] New feature (non-breaking change which adds functionality)

omeranha committed:
Add forge modifications before rebase (thanks for @murilo09)

Rebase was executed to clear the commit list

aokisama commented:
Influenced and Fiendish monsters don't seem to be spawning (no way to get dust)

Also need the action to open the forge, otherwise if you use the button it will say you need to be near exaltation forge to be able to do anything.

I copied the market code and placed this under scripts/actions/object/forge.lua

```
local forge = Action()

function forge.onUse(player, item, fromPosition, target, toPosition, isHotkey)
	player:openForge()
	return true
end

forge:id(37122, 37131, 37128, 37158, 37159, 37155, 37129)
forge:register()
```

murilo09 commented:
@aokisama This PR is still under development, some parts haven't yet been commited (Some Lua files). Thanks for your report.

marcosvf132 commented:
Realy nice PR, looking very sharp!

 There is only one thing that i would change that is the static variables inside the protocolgame. Didn't add it as a review because the compliment should stay if you change it.
I mean those parts right here, and also a few prints that i asume are there when you were testing it. to see if it's working properly.
![image](https://user-images.githubusercontent.com/66353315/194673002-893437bb-5142-4081-a2a6-75d183a8756e.png)

My suggestion, if i may, is to move those static values to the config.lua, so we can have more control about then when customizing and editing the system. I see that it still draft, im leaving my opinion here just as a tip for improvement before opening it as ready for review.

Thank you for your PR, looking really good!!

KaQn commented:
![Capture](https://user-images.githubusercontent.com/77080600/195409869-7728a0f3-55ff-46a2-8c4d-f84172dfcd35.PNG)

Clicking on the items in the market is making the client debug

I believe the error is happening on protocolgame.cpp somewhere in this code

```
if (tier > 0)
	{
		skills_t skillExaltationForge = SKILL_FIRST;
		float bonusValueExaltationForge = 0;

		std::ostringstream ss;
		ss << static_cast<uint16_t>(tier) << " (" << std::setprecision(2) << std::fixed << (bonusValueExaltationForge) << "% " << getSkillName(skillExaltationForge) << ")";
		msg.addString(ss.str()); // Tier
	}
	else
	{
		msg.add<uint16_t>(0); // Tier
	}
```

I was able to select tier 0 items by removing `else
	{
		msg.add<uint16_t>(0); // Tier
	}`
	
but still debugs if I select a tier 1 or higher item, not sure exactly what's wrong. Hope the info helps.

murilo09 commented:
@KaQn I will investigate. Thanks for your report!

KaQn commented:
> @KaQn I will investigate. Thanks for your report!

Thank you for the quick reply, after looking a little more into it, seems like a pretty simple issue, removal of this ```// Item tier modifier (12.82)
	msg.add<uint16_t>(0x00);``` from protocolgame.cpp, market seems to be working fine after taking it off.
	
Okay so buying it works properly after removing that line, but for example I bought an Albino Plate Tier 0 and upon purchase it was Tier 48, with a high chance to proc the effect.

![image](https://user-images.githubusercontent.com/77080600/195429862-e5a47378-e63b-4316-a0a2-012cf626b88c.png)

murilo09 commented:
@KaQn Every issue you have with market, please test it in PR #366 as well. If it's happening in that PR too, comment there. I just ran into this market error too, but I can't test it properly now.

QuebradaZN commented:
Crash Server, Teste executado da seguinte forma, 2 UNICOS item "cobra axe" Tier 1, Depois de upar para Tier 2, ele continua mostrando os 2 items, só ficar clicando como se fosse forjar novamente, fecha a Fusão...

https://cdn.discordapp.com/attachments/699257509252431936/1030892243285983312/WhatsApp_Video_2022-10-15_at_14.16.45.mp4

murilo09 commented:
@QuebradaZN I already knew about this error, it's on the list to fix. Thanks for your report though.

QuebradaZN commented:
When forging, moving an item to Tier 1 for example, If the first attempt fails and without closing the exaltation forge, you forge the same item to tier 1 even if it fails or succeeds the second time, the item doubles!

tutbarao commented:
I did all the compilation but I can't collect "dust". How to do? I can't find mobs with dust

QuebradaZN commented:
> 

If you want dust to test, add it from the database

fellipe005 commented:
I did a basic command to help in testing a god command to add dusts to the account
local tibiadust = TalkAction("/dust")

function tibiadust.onSay(player, words, param)    

    if not player:getGroup():getAccess() or player:getAccountType() < ACCOUNT_TYPE_GOD then
        return true
    end
    local usage = "/dust PLAYER NAME,DUST AMOUNT"
    if param == "" then
        player:sendCancelMessage("Command param required. Usage: ".. usage)
        return false
    end

    local split = param:split(",")
    if not split[2] then
        player:sendCancelMessage("Insufficient parameters. Usage: ".. usage)
        return false
    end

    local target = Player(split[1])
    if not target then
        player:sendCancelMessage("A player with that name is not online.")
        return false
    end
    --trim left
    split[2] = split[2]:gsub("^%s*(.-)$", "%1")
    
    player:sendCancelMessage("Added " .. split[2] .. " tibia dust to the character '" .. target:getName() .. "'.")
    target:sendCancelMessage("Received " .. split[2] .. " tibia dust!")
    target:addForgeDusts(tonumber(split[2]))
    target:getPosition():sendMagicEffect(CONST_ME_HOLYAREA)


end

tibiadust:separator(" ")
tibiadust:register()

fellipe005 commented:
Bom pelo que testei aqui ta funcionando perfeitamente apenas alguns bugs que achei. 
Bugs:
e a fusion um erro da fusion é que quando vc vai forja 2 itens de tier 1 se vc tiver um item de tier 0 na Bp ele puxa um item tier 1 e o item tier 0 em vez de puxar os dois tier 1 para forja um tier 2 e um outro erro no fusion é vc  forja dois tier 8 se ele não forja ele te devolve 1 item com tier 0 e o correto seria devolver um item com o tier 8 e o outro  item com o tier a menos no caso 7 
bom não sei se deu pra explicar tudo mais é isso espero que ajude.

murilo09 committed:
minor fixes

dudantas committed:
Merge branch 'main' into pr/543

dudantas committed:
Add include missing and refactor code complexity

murilo09 committed:
sonar fixes

bennyhdz commented:
Any idea how make spell exiva moe res

dudantas commented on src/server/network/protocol/protocolgame.cpp:
```diff
@@ -6458,12 +6685,44 @@ void ProtocolGame::AddCreature(NetworkMessage &msg, const Creature *creature, bo
+				if (icon != CREATUREICON_NONE) {
+
+					// Icones com stack (Fiendishs e Influenceds)
+					if (monster->getForgeStack() > 0) {
```
> `if (monster->getForgeStack() > 0) {`
> 
> Não sei ao certo se ficou exato

Qual o problema?

dudantas committed:
Merge branch 'main' into pr/543

dudantas committed:
Fix conflict

dudantas committed:
Merge branch 'forge' of https://github.com/murilo09/canary into pr/543

dudantas commented:
> Foram adicionado novas estuturas de pasta e organização das pastas, como fazemos pra atualizar e realizar testes?

Já atualizei o pr, fique a vontade para testar

dudantas committed:
Add final return value

Some forge fixes

Fixed not to remove imbued and equipped items:
Created function getForgeItemFromId... to return only backpack items (ignoring imbued)
Added some nullptr check and logs

dudantas commented:
> Atualizando sobre os testes ao compilar.

Ok, já mandei a correção

QuebradaZN commented:
![Captura de tela 2022-11-21 122412](https://user-images.githubusercontent.com/47014952/203098883-583bf3b6-7a87-4dde-8ccf-3df7d9bbb352.png)
Utilizando Ubuntu 22
Arquitetura Arm

dudantas committed:
Rework on lua enums

Extracted enums into separate functions
Reported error in the comment: https://github.com/opentibiabr/canary/pull/543#issuecomment-1322232126

Variable tracking size limit

dudantas committed:
Simplify logic of function for get sliver and core count

dudantas committed:
Some indentations and check fixes

dudantas commented:
> for (auto itemClassification : g_game().getItemsClassifications())

Se puder testar esse commit, possivelmente corrige o problema: https://github.com/opentibiabr/canary/pull/543/commits/0ba2c42c5ea7e89cadefba204fce0301d06280ee

dudantas committed:
Some sonar check fixes and others things

dudantas committed:
Some small fixes

murilo09 committed:
Added lua dust functions and small fix

dudantas committed:
Sonar fix

dudantas committed:
Fix core and sliver count from stash and inventory and add talkaction for open forge

dudantas committed:
Remove unused

dudantas committed:
Rework on sendOpenForge function

Added the transfer structure and modified some things to make the function easier to understand
Some debugs are still running and need to be investigated, but the window is opening normally

murilo09 committed:
small fixes

dudantas committed:
Indentation

dudantas committed:
Fix debug of tier different of classification

dudantas committed:
Fix transfer get items informations

dudantas committed:
Remove whitespace

murilo09 committed:
Transfer bytes fixed and some checks

murilo09 committed:
some changes in conversion

dudantas committed:
Reorganizing forge functions

Added to addGameTask function and some others refactors

dudantas committed:
Merge branch 'main' into pr/543

dudantas committed:
Add ForgeHistory and some adjustments

dudantas committed:
Some small fixes

murilo09 committed:
Add forge_history table

guispiller commented:
Tested here, and my initial considerations are:
- Need to create migrations file for forge history;
- Transfer is not discounting the correct value from player's balance;
- Bonus message when fusion failed and tier not reduced is not being triggered;
- When exalted cores are at the stash, fusion/transfer is permitted but you lose all the items in the process;
- When slivers are at the stash, exalted core creation fails

murilo09 committed:
fixing reported bugs and history

- Fixing bugs reported in PR
- Some adjusts in history

murilo09 committed:
Fix

murilo09 committed:
History adjusts and sonar fixes

dudantas committed:
Fix return true

murilo09 committed:
some history fixes

dudantas committed:
Merge branch 'main' into forge

dudantas committed:
Fix conflicts

dudantas committed:
Indent to tab

dudantas committed:
Rework on systems exhaustion

Removed individual exhaustions and added one for all systems, as no player will open multiple windows at once, always one at a time.

dudantas committed:
Fix compilation

dudantas committed:
Move check to Game

murilo09 committed:
small remake sendForgeHistory logic

dudantas committed:
Fix history crash

dudantas commented:
> Sobre a transfer, ainda não está aparecendo para transferir o tier! ![image](https://user-images.githubusercontent.com/82473579/205207966-89bca3ce-cfd1-438f-9143-59762d1192d9.png) Haa conseguir fazer

Você não tá clicando em cima do item que vai receber o tier

> Não sei se esta certo, mas quando eu transfiro tier 2 para tier 0 , ele transforma tier 1...

Sim, está correto, quando transfere tier perde um nível

murilo09 committed:
Fix history crash when empty

and more precise formula for Ruse

QuebradaZN commented:
![image](https://user-images.githubusercontent.com/47014952/205210232-9fce4838-ec00-42a6-b83d-143ae57ce091.png)
Deve selecionar o Item que vai receber o Tier

murilo09 commented:
@thunerm o transfer tem algumas condições específicas para funcionar, você pode ler mais aqui: https://tibia.fandom.com/wiki/Equipment_Upgrade#Transferring

dudantas committed:
Fix to remove items from stash

dudantas committed:
Add sendForgeError for send error to player and close forge window

dudantas committed:
Fix amount remove

dudantas committed:
Fix typo

omeranha committed:
fix quiver issues

tutbarao commented:
I tested this point and the demonized creatures still have the same damage and health as a normal creature.

"They can have from 1 to 5 stacks, which give creatures more damage and maximum health - the more they have, the stronger they are and the greater the potential for them to carry more dust with them."

murilo09 commented:
@tutbarao
```
11:33 You see a memory of an amazon.
Health: 4032 / 4032.
Position: 33447, 31063, 12
11:33 You see a memory of an amazon.
Health: 4248 / 4248.
Position: 33449, 31063, 12
11:33 You see a memory of an amazon.
Health: 4680 / 4680.
Position: 33450, 31060, 12
11:33 You see a memory of an amazon.
Health: 6840 / 6840.
Position: 33447, 31060, 12
```

omeranha committed:
fixing nullptr

murilo09 committed:
Set precision to ruse

dudantas committed:
Create new classes to load/save player in iologindata

dudantas committed:
Remove not needed

dudantas committed:
Make variable unused and remove logs

dudantas committed:
Merge branch 'main' into pr/543

dudantas committed:
Merge branch 'main' into pr/543

dudantas commented:
> ![image](https://user-images.githubusercontent.com/82473579/205471716-7d144606-c524-4122-9334-b84243e249f8.png) apareceu isso na distro

Test https://github.com/opentibiabr/canary/pull/663, please

dudantas committed:
Some sonar check fixes and others things

dudantas commented:
> > ![image](https://user-images.githubusercontent.com/82473579/205418197-7f4a8c6a-69b2-4ca1-9d11-95c040d6f3c6.png) bug craching server.
> > mentioned this pull resquest [fix quiver issues](https://github.com/opentibiabr/canary/pull/543/commits/dd40cc004a7774c053fc9eeb9a1d34f40b4c08c2)
> 
> up

I just made a commit where I removed that part of the quiver from the forge pr and put it in another pr, in this one: https://github.com/opentibiabr/canary/pull/622/files

dudantas committed:
Remove not needed

murilo09 committed:
fix sonar issues

dudantas committed:
Fix sonar

dudantas committed:
Fix find fiendish spell only for unlocked bestiary

dudantas committed:
Indent to tab

dudantas committed:
Add forge script in talkactions

dudantas committed:
Add open forge to forge functions

dudantas committed:
Migrate to canary data

dudantas committed:
Change to call parseSendResourceBalance

dudantas committed:
Final fixes

dudantas commented:
> 

O que seria esse drop? Não entendi

dudantas committed:
Fix lag on item creation

murilo09 committed:
add format price function

murilo09 committed:
Some fixes

- Sonar
- Wrong max tier
- Create item duplication

murilo09 committed:
sonar fixes

dudantas committed:
Fix to formatPrice

dudantas committed:
Some fixes

omeranha committed:
send fatal, dodge and momentum chances on cyclopedia and skills window

dudantas committed:
Some fixes

dudantas committed:
Set fiendish only for monster with loot

dudantas committed:
Reorder to public

dudantas committed:
Merge branch 'main' into forge

dudantas committed:
Some fixes and adjustments

Added setting the time it takes to change the fiendish monster in config.lua
Added to create monster with fiendish/influenced in talkaction /m and created a talkaction to set fiendish on monsters
Some fixes to existing functions

dudantas committed:
Fix to check nil

dudantas committed:
Fiendish and influenced monsters true random and others

Fixed some lua scripts and added unordered_set to randomic monsters

dudantas commented:
> ![image](https://user-images.githubusercontent.com/82473579/205756912-4bc0932b-2412-48c3-b48d-8a6cb64300dd.png)

Fixed in last commit.

dudantas committed:
Add trim left

dudantas committed:
Fix build?

dudantas committed:
Remove test log

dudantas committed:
Merge branch 'main' into pr/543

dudantas commented:
> 2022-12-06 00:01:37 -  futex_wait_cancelable (private=0, expected=0, futex_word=0xaaaaaba9f978 <g_loaderSignal+40>) at ../sysdeps/nptl/futex-internal.h:183
> 2022-12-06 00:01:37 -  183	../sysdeps/nptl/futex-internal.h: No such file or directory.
> 2022-12-06 00:01:37 -  
> 2022-12-06 00:01:37 -  Thread 4 (Thread 0xfffff6a591b0 (LWP 929852) "canary"):
> 2022-12-06 00:01:37 -  #0  futex_wait_cancelable (private=0, expected=0, futex_word=0xaaaaaba87de0 <DatabaseTasks::getInstance()::instance+144>) at ../sysdeps/nptl/futex-internal.h:183
> 2022-12-06 00:01:37 -          _x2tmp = 0
> 2022-12-06 00:01:37 -          _x0tmp = 187650001108448
> 2022-12-06 00:01:37 -          _x0 = 187650001108448
> 2022-12-06 00:01:37 -          _x2 = 0
> 2022-12-06 00:01:37 -          _x8 = 98
> 2022-12-06 00:01:37 -          _x3tmp = 0
> 2022-12-06 00:01:37 -          _x1tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x1 = 128
> 2022-12-06 00:01:37 -          _x3 = 0
> 2022-12-06 00:01:37 -          _sys_result = <optimized out>
> 2022-12-06 00:01:37 -          __ret = <optimized out>
> 2022-12-06 00:01:37 -          oldtype = 0
> 2022-12-06 00:01:37 -          err = <optimized out>
> 2022-12-06 00:01:37 -          oldtype = <optimized out>
> 2022-12-06 00:01:37 -          err = <optimized out>
> 2022-12-06 00:01:37 -          __ret = <optimized out>
> 2022-12-06 00:01:37 -          _sys_result = <optimized out>
> 2022-12-06 00:01:37 -          _x3tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x2tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x1tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x0tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x0 = <optimized out>
> 2022-12-06 00:01:37 -          _x1 = <optimized out>
> 2022-12-06 00:01:37 -          _x2 = <optimized out>
> 2022-12-06 00:01:37 -          _x3 = <optimized out>
> 2022-12-06 00:01:37 -          _x8 = <optimized out>
> 2022-12-06 00:01:37 -  #1  __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0xaaaaaba87d88 <DatabaseTasks::getInstance()::instance+56>, cond=0xaaaaaba87db8 <DatabaseTasks::getInstance()::instance+104>) at pthread_cond_wait.c:508
> 2022-12-06 00:01:37 -          spin = 0
> 2022-12-06 00:01:37 -          buffer = {__routine = 0xfffff7fa9158 <__condvar_cleanup_waiting>, __arg = 0xfffff6a587d8, __canceltype = -1428881612, __prev = 0x0}
> 2022-12-06 00:01:37 -          cbuffer = {wseq = 0, cond = 0xaaaaaba87db8 <DatabaseTasks::getInstance()::instance+104>, mutex = 0xaaaaaba87d88 <DatabaseTasks::getInstance()::instance+56>, private = 0}
> 2022-12-06 00:01:37 -          err = <optimized out>
> 2022-12-06 00:01:37 -          g = 0
> 2022-12-06 00:01:37 -          flags = <optimized out>
> 2022-12-06 00:01:37 -          g1_start = <optimized out>
> 2022-12-06 00:01:37 -          maxspin = 0
> 2022-12-06 00:01:37 -          signals = <optimized out>
> 2022-12-06 00:01:37 -          result = 0
> 2022-12-06 00:01:37 -          wseq = 0
> 2022-12-06 00:01:37 -          seq = 0
> 2022-12-06 00:01:37 -          private = 0
> 2022-12-06 00:01:37 -          maxspin = <optimized out>
> 2022-12-06 00:01:37 -          err = <optimized out>
> 2022-12-06 00:01:37 -          result = <optimized out>
> 2022-12-06 00:01:37 -          wseq = <optimized out>
> 2022-12-06 00:01:37 -          g = <optimized out>
> 2022-12-06 00:01:37 -          seq = <optimized out>
> 2022-12-06 00:01:37 -          flags = <optimized out>
> 2022-12-06 00:01:37 -          private = <optimized out>
> 2022-12-06 00:01:37 -          signals = <optimized out>
> 2022-12-06 00:01:37 -          done = <optimized out>
> 2022-12-06 00:01:37 -          g1_start = <optimized out>
> 2022-12-06 00:01:37 -          spin = <optimized out>
> 2022-12-06 00:01:37 -          buffer = {__routine = <optimized out>, __arg = <optimized out>, __canceltype = <optimized out>, __prev = <optimized out>}
> 2022-12-06 00:01:37 -          cbuffer = {wseq = <optimized out>, cond = <optimized out>, mutex = <optimized out>, private = <optimized out>}
> 2022-12-06 00:01:37 -          s = <optimized out>
> 2022-12-06 00:01:37 -  #2  __pthread_cond_wait (cond=0xaaaaaba87db8 <DatabaseTasks::getInstance()::instance+104>, mutex=0xaaaaaba87d88 <DatabaseTasks::getInstance()::instance+56>) at pthread_cond_wait.c:647
> 2022-12-06 00:01:37 -  No locals.
> 2022-12-06 00:01:37 -  #3  0x0000fffff7d8bab0 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /lib/aarch64-linux-gnu/libstdc++.so.6
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #4  0x0000aaaaaad4f610 in DatabaseTasks::threadMain() ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #5  0x0000aaaaaad520cc in void std::__invoke_impl<void, void (DatabaseTasks::*)(), DatabaseTasks*>(std::__invoke_memfun_deref, void (DatabaseTasks::*&&)(), DatabaseTasks*&&) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #6  0x0000aaaaaad51fe4 in std::__invoke_result<void (DatabaseTasks::*)(), DatabaseTasks*>::type std::__invoke<void (DatabaseTasks::*)(), DatabaseTasks*>(void (DatabaseTasks::*&&)(), DatabaseTasks*&&) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #7  0x0000aaaaaad51f4c in void std::thread::_Invoker<std::tuple<void (DatabaseTasks::*)(), DatabaseTasks*> >::_M_invoke<0ul, 1ul>(std::_Index_tuple<0ul, 1ul>) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #8  0x0000aaaaaad51f08 in std::thread::_Invoker<std::tuple<void (DatabaseTasks::*)(), DatabaseTasks*> >::operator()() ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #9  0x0000aaaaaad51eec in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (DatabaseTasks::*)(), DatabaseTasks*> > >::_M_run() ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #10 0x0000fffff7d9270c in ?? () from /lib/aarch64-linux-gnu/libstdc++.so.6
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #11 0x0000fffff7fa2624 in start_thread (arg=0xfffff7d926f0) at pthread_create.c:477
> 2022-12-06 00:01:37 -          ret = <optimized out>
> 2022-12-06 00:01:37 -          pd = <optimized out>
> 2022-12-06 00:01:37 -          unwind_buf = {cancel_jmp_buf = {{jmp_buf = {281474819790284, 281474836712384, 281474836712382, 281474842255360, 281474836712383, 281474839946992, 281474819791024, 281474819789232, 281474842259456, 281474819789232, 281474819787152, 6782860774225554561, 0, 6782860774247923725, 0, 0, 0, 0, 0, 0, 0, 0}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
> 2022-12-06 00:01:37 -          not_first_call = <optimized out>
> 2022-12-06 00:01:37 -  #12 0x0000fffff7bfb49c in thread_start () at ../sysdeps/unix/sysv/linux/aarch64/clone.S:78
> 2022-12-06 00:01:37 -  No locals.
> 2022-12-06 00:01:37 -  
> 2022-12-06 00:01:37 -  Thread 3 (Thread 0xfffff727d1b0 (LWP 929851) "canary"):
> 2022-12-06 00:01:37 -  #0  futex_wait_cancelable (private=0, expected=0, futex_word=0xaaaaaba99bd0 <Scheduler::getInstance()::instance+112>) at ../sysdeps/nptl/futex-internal.h:183
> 2022-12-06 00:01:37 -          _x2tmp = 0
> 2022-12-06 00:01:37 -          _x0tmp = 187650001181648
> 2022-12-06 00:01:37 -          _x0 = 187650001181648
> 2022-12-06 00:01:37 -          _x2 = 0
> 2022-12-06 00:01:37 -          _x8 = 98
> 2022-12-06 00:01:37 -          _x3tmp = 0
> 2022-12-06 00:01:37 -          _x1tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x1 = 128
> 2022-12-06 00:01:37 -          _x3 = 0
> 2022-12-06 00:01:37 -          _sys_result = <optimized out>
> 2022-12-06 00:01:37 -          __ret = <optimized out>
> 2022-12-06 00:01:37 -          oldtype = 0
> 2022-12-06 00:01:37 -          err = <optimized out>
> 2022-12-06 00:01:37 -          oldtype = <optimized out>
> 2022-12-06 00:01:37 -          err = <optimized out>
> 2022-12-06 00:01:37 -          __ret = <optimized out>
> 2022-12-06 00:01:37 -          _sys_result = <optimized out>
> 2022-12-06 00:01:37 -          _x3tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x2tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x1tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x0tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x0 = <optimized out>
> 2022-12-06 00:01:37 -          _x1 = <optimized out>
> 2022-12-06 00:01:37 -          _x2 = <optimized out>
> 2022-12-06 00:01:37 -          _x3 = <optimized out>
> 2022-12-06 00:01:37 -          _x8 = <optimized out>
> 2022-12-06 00:01:37 -  #1  __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0xaaaaaba99b78 <Scheduler::getInstance()::instance+24>, cond=0xaaaaaba99ba8 <Scheduler::getInstance()::instance+72>) at pthread_cond_wait.c:508
> 2022-12-06 00:01:37 -          spin = 0
> 2022-12-06 00:01:37 -          buffer = {__routine = 0xfffff7fa9158 <__condvar_cleanup_waiting>, __arg = 0xfffff727c7e8, __canceltype = -1414947824, __prev = 0x0}
> 2022-12-06 00:01:37 -          cbuffer = {wseq = 0, cond = 0xaaaaaba99ba8 <Scheduler::getInstance()::instance+72>, mutex = 0xaaaaaba99b78 <Scheduler::getInstance()::instance+24>, private = 0}
> 2022-12-06 00:01:37 -          err = <optimized out>
> 2022-12-06 00:01:37 -          g = 0
> 2022-12-06 00:01:37 -          flags = <optimized out>
> 2022-12-06 00:01:37 -          g1_start = <optimized out>
> 2022-12-06 00:01:37 -          maxspin = 0
> 2022-12-06 00:01:37 -          signals = <optimized out>
> 2022-12-06 00:01:37 -          result = 0
> 2022-12-06 00:01:37 -          wseq = 0
> 2022-12-06 00:01:37 -          seq = 0
> 2022-12-06 00:01:37 -          private = 0
> 2022-12-06 00:01:37 -          maxspin = <optimized out>
> 2022-12-06 00:01:37 -          err = <optimized out>
> 2022-12-06 00:01:37 -          result = <optimized out>
> 2022-12-06 00:01:37 -          wseq = <optimized out>
> 2022-12-06 00:01:37 -          g = <optimized out>
> 2022-12-06 00:01:37 -          seq = <optimized out>
> 2022-12-06 00:01:37 -          flags = <optimized out>
> 2022-12-06 00:01:37 -          private = <optimized out>
> 2022-12-06 00:01:37 -          signals = <optimized out>
> 2022-12-06 00:01:37 -          done = <optimized out>
> 2022-12-06 00:01:37 -          g1_start = <optimized out>
> 2022-12-06 00:01:37 -          spin = <optimized out>
> 2022-12-06 00:01:37 -          buffer = {__routine = <optimized out>, __arg = <optimized out>, __canceltype = <optimized out>, __prev = <optimized out>}
> 2022-12-06 00:01:37 -          cbuffer = {wseq = <optimized out>, cond = <optimized out>, mutex = <optimized out>, private = <optimized out>}
> 2022-12-06 00:01:37 -          s = <optimized out>
> 2022-12-06 00:01:37 -  #2  __pthread_cond_wait (cond=0xaaaaaba99ba8 <Scheduler::getInstance()::instance+72>, mutex=0xaaaaaba99b78 <Scheduler::getInstance()::instance+24>) at pthread_cond_wait.c:647
> 2022-12-06 00:01:37 -  No locals.
> 2022-12-06 00:01:37 -  #3  0x0000fffff7d8bab0 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /lib/aarch64-linux-gnu/libstdc++.so.6
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #4  0x0000aaaaaadc5594 in Scheduler::threadMain() ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #5  0x0000aaaaab000e38 in void std::__invoke_impl<void, void (Scheduler::*)(), Scheduler*>(std::__invoke_memfun_deref, void (Scheduler::*&&)(), Scheduler*&&) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #6  0x0000aaaaab000c3c in std::__invoke_result<void (Scheduler::*)(), Scheduler*>::type std::__invoke<void (Scheduler::*)(), Scheduler*>(void (Scheduler::*&&)(), Scheduler*&&) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #7  0x0000aaaaab000940 in void std::thread::_Invoker<std::tuple<void (Scheduler::*)(), Scheduler*> >::_M_invoke<0ul, 1ul>(std::_Index_tuple<0ul, 1ul>) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #8  0x0000aaaaab00061c in std::thread::_Invoker<std::tuple<void (Scheduler::*)(), Scheduler*> >::operator()() ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #9  0x0000aaaaaafffa2c in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (Scheduler::*)(), Scheduler*> > >::_M_run() ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #10 0x0000fffff7d9270c in ?? () from /lib/aarch64-linux-gnu/libstdc++.so.6
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #11 0x0000fffff7fa2624 in start_thread (arg=0xfffff7d926f0) at pthread_create.c:477
> 2022-12-06 00:01:37 -          ret = <optimized out>
> 2022-12-06 00:01:37 -          pd = <optimized out>
> 2022-12-06 00:01:37 -          unwind_buf = {cancel_jmp_buf = {{jmp_buf = {281474828326348, 281474976707248, 281474976707246, 281474842255360, 281474976707247, 281474839946992, 281474828327088, 281474828325296, 281474842259456, 281474828325296, 281474828323216, 6782860774225554561, 0, 6782860774222643213, 0, 0, 0, 0, 0, 0, 0, 0}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
> 2022-12-06 00:01:37 -          not_first_call = <optimized out>
> 2022-12-06 00:01:37 -  #12 0x0000fffff7bfb49c in thread_start () at ../sysdeps/unix/sysv/linux/aarch64/clone.S:78
> 2022-12-06 00:01:37 -  No locals.
> 2022-12-06 00:01:37 -  
> 2022-12-06 00:01:37 -  Thread 2 (Thread 0xfffff7a7e1b0 (LWP 929850) "canary"):
> 2022-12-06 00:01:37 -  #0  0x0000fffff7e02678 in void std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_construct<char*>(char*, char*, std::forward_iterator_tag) () from /lib/aarch64-linux-gnu/libstdc++.so.6
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #1  0x0000aaaaaabbef24 in std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>::pair(std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short> const&) [clone .lto_priv.0] ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #2  0x0000aaaaaae7bb70 in bool __gnu_cxx::__ops::_Iter_pred<Items::parseItemNode(pugi::xml_node const&, unsigned short)::{lambda(auto:1)#1}>::operator()<std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true> >(std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true>) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #3  0x0000aaaaaae7b318 in std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true> std::__find_if<std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true>, __gnu_cxx::__ops::_Iter_pred<Items::parseItemNode(pugi::xml_node const&, unsigned short)::{lambda(auto:1)#1}> >(std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true>, std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true>, __gnu_cxx::__ops::_Iter_pred<Items::parseItemNode(pugi::xml_node const&, unsigned short)::{lambda(auto:1)#1}>, std::input_iterator_tag) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #4  0x0000aaaaaae7ab94 in std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true> std::__find_if<std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true>, __gnu_cxx::__ops::_Iter_pred<Items::parseItemNode(pugi::xml_node const&, unsigned short)::{lambda(auto:1)#1}> >(std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true>, std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true>, __gnu_cxx::__ops::_Iter_pred<Items::parseItemNode(pugi::xml_node const&, unsigned short)::{lambda(auto:1)#1}>) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #5  0x0000aaaaaae79f84 in std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true> std::find_if<std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true>, Items::parseItemNode(pugi::xml_node const&, unsigned short)::{lambda(auto:1)#1}>(std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true>, std::__detail::_Node_iterator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, unsigned short>, false, true>, Items::parseItemNode(pugi::xml_node const&, unsigned short)::{lambda(auto:1)#1}) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #6  0x0000aaaaaae78f94 in Items::parseItemNode(pugi::xml_node const&, unsigned short) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #7  0x0000aaaaaae78a98 in Items::loadFromXml() ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #8  0x0000aaaaaafe8bd0 in loadModules() ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #9  0x0000aaaaaafe9cb8 in mainLoader(int, char**, ServiceManager*) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #10 0x0000aaaaaaff846c in void std::__invoke_impl<void, void (*&)(int, char**, ServiceManager*), int&, char**&, ServiceManager*&>(std::__invoke_other, void (*&)(int, char**, ServiceManager*), int&, char**&, ServiceManager*&) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #11 0x0000aaaaaaff74e8 in std::__invoke_result<void (*&)(int, char**, ServiceManager*), int&, char**&, ServiceManager*&>::type std::__invoke<void (*&)(int, char**, ServiceManager*), int&, char**&, ServiceManager*&>(void (*&)(int, char**, ServiceManager*), int&, char**&, ServiceManager*&) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #12 0x0000aaaaaaff6404 in void std::_Bind<void (*(int, char**, ServiceManager*))(int, char**, ServiceManager*)>::__call<void, , 0ul, 1ul, 2ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul, 2ul>) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #13 0x0000aaaaaaff5318 in void std::_Bind<void (*(int, char**, ServiceManager*))(int, char**, ServiceManager*)>::operator()<, void>() ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #14 0x0000aaaaaaff4614 in void std::__invoke_impl<void, std::_Bind<void (*(int, char**, ServiceManager*))(int, char**, ServiceManager*)>&>(std::__invoke_other, std::_Bind<void (*(int, char**, ServiceManager*))(int, char**, ServiceManager*)>&) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #15 0x0000aaaaaaff2c90 in std::enable_if<is_invocable_r_v<void, std::_Bind<void (*(int, char**, ServiceManager*))(int, char**, ServiceManager*)>&>, void>::type std::__invoke_r<void, std::_Bind<void (*(int, char**, ServiceManager*))(int, char**, ServiceManager*)>&>(std::_Bind<void (*(int, char**, ServiceManager*))(int, char**, ServiceManager*)>&) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #16 0x0000aaaaaafefdc4 in std::_Function_handler<void (), std::_Bind<void (*(int, char**, ServiceManager*))(int, char**, ServiceManager*)> >::_M_invoke(std::_Any_data const&) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #17 0x0000aaaaaadca714 in std::function<void ()>::operator()() const ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #18 0x0000aaaaaadc9f64 in Task::operator()() ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #19 0x0000aaaaaadca1b0 in Dispatcher::threadMain() ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #20 0x0000aaaaab000f18 in void std::__invoke_impl<void, void (Dispatcher::*)(), Dispatcher*>(std::__invoke_memfun_deref, void (Dispatcher::*&&)(), Dispatcher*&&) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #21 0x0000aaaaab000cd8 in std::__invoke_result<void (Dispatcher::*)(), Dispatcher*>::type std::__invoke<void (Dispatcher::*)(), Dispatcher*>(void (Dispatcher::*&&)(), Dispatcher*&&) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #22 0x0000aaaaab000988 in void std::thread::_Invoker<std::tuple<void (Dispatcher::*)(), Dispatcher*> >::_M_invoke<0ul, 1ul>(std::_Index_tuple<0ul, 1ul>) ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #23 0x0000aaaaab000638 in std::thread::_Invoker<std::tuple<void (Dispatcher::*)(), Dispatcher*> >::operator()() ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #24 0x0000aaaaaafffa4c in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (Dispatcher::*)(), Dispatcher*> > >::_M_run() ()
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #25 0x0000fffff7d9270c in ?? () from /lib/aarch64-linux-gnu/libstdc++.so.6
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #26 0x0000fffff7fa2624 in start_thread (arg=0xfffff7d926f0) at pthread_create.c:477
> 2022-12-06 00:01:37 -          ret = <optimized out>
> 2022-12-06 00:01:37 -          pd = <optimized out>
> 2022-12-06 00:01:37 -          unwind_buf = {cancel_jmp_buf = {{jmp_buf = {281474836719052, 281474976707248, 281474976707246, 281474842255360, 281474976707247, 281474839946992, 281474836719792, 281474836718000, 281474842259456, 281474836718000, 281474836715920, 6782860774225554561, 0, 6782860774231035917, 0, 0, 0, 0, 0, 0, 0, 0}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
> 2022-12-06 00:01:37 -          not_first_call = <optimized out>
> 2022-12-06 00:01:37 -  #27 0x0000fffff7bfb49c in thread_start () at ../sysdeps/unix/sysv/linux/aarch64/clone.S:78
> 2022-12-06 00:01:37 -  No locals.
> 2022-12-06 00:01:37 -  
> 2022-12-06 00:01:37 -  Thread 1 (Thread 0xfffff7a91010 (LWP 929847) "canary"):
> 2022-12-06 00:01:37 -  #0  futex_wait_cancelable (private=0, expected=0, futex_word=0xaaaaaba9f978 <g_loaderSignal+40>) at ../sysdeps/nptl/futex-internal.h:183
> 2022-12-06 00:01:37 -          _x2tmp = 0
> 2022-12-06 00:01:37 -          _x0tmp = 187650001205624
> 2022-12-06 00:01:37 -          _x0 = 187650001205624
> 2022-12-06 00:01:37 -          _x2 = 0
> 2022-12-06 00:01:37 -          _x8 = 98
> 2022-12-06 00:01:37 -          _x3tmp = 0
> 2022-12-06 00:01:37 -          _x1tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x1 = 128
> 2022-12-06 00:01:37 -          _x3 = 0
> 2022-12-06 00:01:37 -          _sys_result = <optimized out>
> 2022-12-06 00:01:37 -          __ret = <optimized out>
> 2022-12-06 00:01:37 -          oldtype = 0
> 2022-12-06 00:01:37 -          err = <optimized out>
> 2022-12-06 00:01:37 -          oldtype = <optimized out>
> 2022-12-06 00:01:37 -          err = <optimized out>
> 2022-12-06 00:01:37 -          __ret = <optimized out>
> 2022-12-06 00:01:37 -          _sys_result = <optimized out>
> 2022-12-06 00:01:37 -          _x3tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x2tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x1tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x0tmp = <optimized out>
> 2022-12-06 00:01:37 -          _x0 = <optimized out>
> 2022-12-06 00:01:37 -          _x1 = <optimized out>
> 2022-12-06 00:01:37 -          _x2 = <optimized out>
> 2022-12-06 00:01:37 -          _x3 = <optimized out>
> 2022-12-06 00:01:37 -          _x8 = <optimized out>
> 2022-12-06 00:01:37 -  #1  __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0xaaaaaba9f920 <g_loaderLock>, cond=0xaaaaaba9f950 <g_loaderSignal>) at pthread_cond_wait.c:508
> 2022-12-06 00:01:37 -          spin = 0
> 2022-12-06 00:01:37 -          buffer = {__routine = 0xfffff7fa9158 <__condvar_cleanup_waiting>, __arg = 0xfffffffff398, __canceltype = -1414948044, __prev = 0x0}
> 2022-12-06 00:01:37 -          cbuffer = {wseq = 0, cond = 0xaaaaaba9f950 <g_loaderSignal>, mutex = 0xaaaaaba9f920 <g_loaderLock>, private = 0}
> 2022-12-06 00:01:37 -          err = <optimized out>
> 2022-12-06 00:01:37 -          g = 0
> 2022-12-06 00:01:37 -          flags = <optimized out>
> 2022-12-06 00:01:37 -          g1_start = <optimized out>
> 2022-12-06 00:01:37 -          maxspin = 0
> 2022-12-06 00:01:37 -          signals = <optimized out>
> 2022-12-06 00:01:37 -          result = 0
> 2022-12-06 00:01:37 -          wseq = 0
> 2022-12-06 00:01:37 -          seq = 0
> 2022-12-06 00:01:37 -          private = 0
> 2022-12-06 00:01:37 -          maxspin = <optimized out>
> 2022-12-06 00:01:37 -          err = <optimized out>
> 2022-12-06 00:01:37 -          result = <optimized out>
> 2022-12-06 00:01:37 -          wseq = <optimized out>
> 2022-12-06 00:01:37 -          g = <optimized out>
> 2022-12-06 00:01:37 -          seq = <optimized out>
> 2022-12-06 00:01:37 -          flags = <optimized out>
> 2022-12-06 00:01:37 -          private = <optimized out>
> 2022-12-06 00:01:37 -          signals = <optimized out>
> 2022-12-06 00:01:37 -          done = <optimized out>
> 2022-12-06 00:01:37 -          g1_start = <optimized out>
> 2022-12-06 00:01:37 -          spin = <optimized out>
> 2022-12-06 00:01:37 -          buffer = {__routine = <optimized out>, __arg = <optimized out>, __canceltype = <optimized out>, __prev = <optimized out>}
> 2022-12-06 00:01:37 -          cbuffer = {wseq = <optimized out>, cond = <optimized out>, mutex = <optimized out>, private = <optimized out>}
> 2022-12-06 00:01:37 -          s = <optimized out>
> 2022-12-06 00:01:37 -  #2  __pthread_cond_wait (cond=0xaaaaaba9f950 <g_loaderSignal>, mutex=0xaaaaaba9f920 <g_loaderLock>) at pthread_cond_wait.c:647
> 2022-12-06 00:01:37 -  No locals.
> 2022-12-06 00:01:37 -  #3  0x0000fffff7d8bab0 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /lib/aarch64-linux-gnu/libstdc++.so.6
> 2022-12-06 00:01:37 -  No symbol table info available.
> 2022-12-06 00:01:37 -  #4  0x0000aaaaaafe962c in main ()
> 2022-12-06 00:01:37 -  No symbol table info available.

How did you reproduce this? The log doesn't help at all if you don't tell how it happened.

murilo09 committed:
somes fixes

dudantas committed:
Add lib magic-enum for use with enum class and fix randomic monsters creation

Change to enum class and add magic enum lib

Fix randomic monsters creation

carlospess0a committed:
Add tier stats to Skills tab and Cyclopedia (#1)

dudantas committed:
Fix crash and remove log

dudantas committed:
Fix influenced talkaction for add new influenced with /m talkaction

dudantas committed:
Merge branch 'main' into forge

dudantas committed:
Merge branch 'main' into pr/543

dudantas committed:
Fix conflicts

dudantas committed:
Merge branch 'forge' of https://github.com/murilo09/canary into pr/543

dudantas committed:
Fix compilation

dudantas committed:
Fix sonar code smell

dudantas committed:
Fixed for not appear item with imbuement in forge window

dudantas committed:
Merge branch 'main' into pr/543

dudantas reviewed (approved):
A great system, well tested and well-developed, shows that teamwork pays off. 🚀

Diego-OT commented:
> A great system, well tested and well-developed, shows that teamwork pays off. 🚀

who did this system?
